src/code/code_80097A00.c

```diff
@@ -11,26 +11,67 @@ u32 gBitFlags[] = {
     (1 << 24), (1 << 25), (1 << 26), (1 << 27), (1 << 28), (1 << 29), (1 << 30), (1 << 31),
 };
 
-u16 gEquipMasks[] = { 0x000F, 0x00F0, 0x0F00, 0xF000 };
-u16 gEquipNegMasks[] = { 0xFFF0, 0xFF0F, 0xF0FF, 0x0FFF };
-u32 gUpgradeMasks[] = {
-    0x00000007, 0x00000038, 0x000001C0, 0x00000E00, 0x00003000, 0x0001C000, 0x000E0000, 0x00700000,
+u16 gEquipMasks[EQUIP_TYPE_MAX] = {
+    0xF << (EQUIP_TYPE_SWORD * 4),  // EQUIP_TYPE_SWORD
+    0xF << (EQUIP_TYPE_SHIELD * 4), // EQUIP_TYPE_SHIELD
+    0xF << (EQUIP_TYPE_TUNIC * 4),  // EQUIP_TYPE_TUNIC
+    0xF << (EQUIP_TYPE_BOOTS * 4),  // EQUIP_TYPE_BOOTS
 };
-u32 gUpgradeNegMasks[] = {
-    0xFFFFFFF8, 0xFFFFFFC7, 0xFFFFFE3F, 0xFFFFF1FF, 0xFFFFCFFF, 0xFFFE3FFF, 0xFFF1FFFF, 0xFF8FFFFF,
+u16 gEquipNegMasks[EQUIP_TYPE_MAX] = {
+    (u16) ~(0xF << (EQUIP_TYPE_SWORD * 4)),  // EQUIP_TYPE_SWORD
+    (u16) ~(0xF << (EQUIP_TYPE_SHIELD * 4)), // EQUIP_TYPE_SHIELD
+    (u16) ~(0xF << (EQUIP_TYPE_TUNIC * 4)),  // EQUIP_TYPE_TUNIC
+    (u16) ~(0xF << (EQUIP_TYPE_BOOTS * 4)),  // EQUIP_TYPE_BOOTS
```

Comment:
i get it was probably done for consistency with the other arrays in the file. but the comments dont feel needed since the enum is right there in the definition

Reply:
I agree partially, imo deriving the corresponding type from the value would just be a (fair) assumption but I feel like alleviating any doubt is better